### ISSUE TITLE
Add Cargo format and clippy checks to our CI for all rust code

### ISSUE THREAD

**Issue by astoycos** [kubernetes-sigs/blixt#212]
Similar to what we do in bpfman at -> https://github.com/bpfman/bpfman/blob/main/.github/workflows/build.yml#L84

we should be running `cargo fmt` and `cargo clippy` against all our rust based code to make sure it's formatted and checked correctly.

**Comment by EandrewJones** [comment]
Happy to take this on per our conversation on slack. To confirm, this looks as simple as adding these two steps to our GH actions for all rust contexts. Based on my perusal, I see the following rust contexts:

```
dataplane/
tools/udp-test-server
```

Do we want this in `ci.yaml` and, if so, one or both of the build steps? Looks like there is one first for all PRs then another for pushes to main.

A manual run of these commands throws warnings -- will fix up those too.

**Comment by shaneutt** [comment]
Yes, and also a note (cc @aryan9600) that we are [rewriting our control-plane into Rust as well](https://github.com/kubernetes-sigs/blixt/milestone/8).

**Comment by EandrewJones** [comment]
I'm tracking the migration. If one of those PRs beats this in, I'll add coverage for those contexts too.

**Comment by qzqsmile** [comment]
As a new contributor, I'm interested in rust and k8s. so can I take this issue?

**Comment by astoycos** [comment]
Hiya @qzqsmile Last I checked @EandrewJones was already looking into this one!

@EandrewJones Can I assign you here?

**Comment by EandrewJones** [comment]
Yes. Please assign.

Best

Evan Jones
Website: www.ea-jones.com
